lambdaclass/mirra_backend
# Campaign rewards

Waiting on #337

Implements level rewards. Rewards can come in the form of currency, units (always level 1 and tier 1) and items (always level 1). Rewards are given automatically when a level is beaten.

Unit ranks will be customizable per reward but we don't have them yet (see: https://github.com/lambdaclass/mirra_backend/pull/329).

# To test

- Create user, get campaigns, get levels

```
{:ok, user} = Champions.Users.register("User")
{:ok, [campaign | _]} = Champions.Campaigns.get_campaigns
[level1, level2, level3 | _] = campaign.levels |> Enum.sort_by(& &1.level_number)

```

- Check out your gold

```
GameBackend.Users.Currencies.get_amount_of_currency(user.id, GameBackend.Users.Currencies.get_currency_by_name!("Gold").id)
```

- Beat level 1 (gives only currency)

```
Champions.Battle.fight_level(user.id, level1.id)
```

- Check out your gold again (should have grown)

```
GameBackend.Users.Currencies.get_amount_of_currency(user.id, GameBackend.Users.Currencies.get_currency_by_name!("Gold").id)
```

- Check your items

```
GameBackend.Users.get_user(user.id) |> elem(1) |> Map.get(:items) |> Enum.count()
```

- Fight level 2 (gives an item)

```
Champions.Battle.fight_level(user.id, level2.id)
```

- Check your items again. Should have grown.

```
GameBackend.Users.get_user(user.id) |> elem(1) |> Map.get(:items) |> Enum.count()
```

- Same for units and level 3
```
Champions.Battle.fight_level(user.id, level3.id)
```

```
GameBackend.Users.get_user(user.id) |> elem(1) |> Map.get(:units) |> Enum.count()
```
